### ISSUE TITLE
ZRTP key continuity feature in CSipSimple?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sonuarya/csipsimple#1153]
```
ZRTP description states that:

...For usability purpose, the SAS can be verified only once, then each party 
can mark the other as “trusted”. In this way the parties do not have to 
verify the SAS in every call. This great feature is provided by the key 
continuity feature of ZRTP...

Are there any chances to implement this functionality in CSipSimple as it is 
VERY important?

It is not difficult to confirm SAS every time if you make encrypted calls time 
to time. 

But suppose you have a workgroup and use SIP for your regular communications on 
daily basis making 30-50 calls a day?.. 

People need to concentrate on their business tasks, not clicking of buttons. 
After very short time they start ignoring that annoying SAS verification thus 
making all your security efforts useless. 

BTW, 
due to similar social factors I fully agree with @jtaylor comment #50 in issue 
262 (http://code.google.com/p/csipsimple/issues/detail?id=262) that we need an 
option to disallow and drop a call if ZRTP session could not be successfully 
established. Or just generate annoying sound tone until ZRTP session is 
established.

Otherwise this is a kind of Russian Roulette when you never can be shure if all 
of your employees calls are being encrypted. Sure, people should check for the 
SAS message every time but imagine making lots of calls every day. Often 
happens that even if both sides have SIP clients with ZRTP enabled ZRTP session 
fails due to networking issues (especially on 3G). But the call is already 
established so people just start talking despite any messages being displayed 
or not.



```

Original issue reported on code.google.com by `d...@dmit.lv` on 15 Jul 2011 at 1:44
- Merged into: #262
